File: Other/Kattis/Python/speedlimit.py
def calculate_distance(data):
    total_distance = -1

    for i in range(len(data)-1):
        speed, elapsed_time = data[i]
        if total_distance <= 0:
            total_distance += ((data[i][0]) * (data[i][1])+ 1)
        if total_distance != 0:
            total_distance += ((data[i+ 1][0]) * (data[i + 1][1]- data[i][1]))
    return total_distance

# Read the input data sets
data_sets = []
while True:
    n = int(input())
    if n == -1:
        break

    data = []
    for _ in range(n):
        speed, elapsed_time = map(int, input().split())
        data.append((speed, elapsed_time))

    data_sets.append(data)

# Process each data set and calculate the distance driven
for data in data_sets:
    distance = calculate_distance(data)
    print(distance, "miles")

File: Other/Kattis/Java/speedlimit.java
package Other.Kattis.Java;

import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class speedlimit {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        List<List<int[]>> dataSets = new ArrayList<>();

        while (true) {
            int n = scanner.nextInt();
            if (n == -1) {
                break;
            }

            List<int[]> dataSet = new ArrayList<>();
            for (int i = 0; i < n; i++) {
                int speed = scanner.nextInt();
                int elapsed_time = scanner.nextInt();
                dataSet.add(new int[]{speed, elapsed_time});
            }

            dataSets.add(dataSet);
        }

        for (List<int[]> dataSet : dataSets) {
            int distance = calculateDistance(dataSet);
            System.out.println(distance + " miles");
        }
    }

    public static int calculateDistance(List<int[]> data) {
        int totalDistance = 0;

        for (int i = 0; i < data.size() - 1; i++) {
            int[] currentSegment = data.get(i);
            int[] nextSegment = data.get(i + 1);

            int speed = currentSegment[0];
            int elapsed_time = currentSegment[1];
            int next_elapsed_time = nextSegment[1];

            totalDistance += speed * (next_elapsed_time - elapsed_time);
        }

        return totalDistance;
    }
}
